docs/azure/sdk/unit-testing-mocking.md

```diff
@@ -0,0 +1,240 @@
+---
+title: Unit testing and mocking with the Azure SDK for .NET
+description: Learn techniques and tools for unit testing and mocking the Azure SDK for .NET
+ms.custom: devx-track-dotnet, engagement-fy23
+ms.date: 07/05/2023
+---
+
+# Unit testing and mocking with the Azure SDK for .NET
+
+Unit testing is an important part of a sustainable development process that can improve code quality and prevent regressions or bugs in your apps. However, unit testing presents challenges when the code you're testing performs network calls, such as those made to Azure resources. Tests that run against live services can experience issues, such as latency that slows down test execution, dependencies on code outside of the isolated test, and issues with managing service state and costs every time the test is run. Instead of testing against live Azure services, replace the service clients with mocked or in-memory implementations. This avoids the above issues and lets developers focus on testing their application logic, independent from the network and service.
+
+In this article, you'll learn how to write unit tests for the Azure SDK for .NET that isolate your dependencies to make your tests more reliable. You'll also learn how to replace key components with in-memory test implementations to create fast and reliable unit tests, and see how to design your own classes to better support unit testing. This article includes examples that use [Moq](https://www.nuget.org/packages/moq/), which is a popular mocking and testing library for .NET.
```

Comment:
Now that we have NSubstitute code samples, mention that library in the last sentence too.